python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/sling/asset_decorator.py

```diff
@@ -88,7 +88,11 @@ def my_assets(context, sling: SlingResource):
                 key=dagster_sling_translator.get_asset_key(stream),
                 deps=dagster_sling_translator.get_deps_asset_key(stream),
                 description=dagster_sling_translator.get_description(stream),
-                metadata=dagster_sling_translator.get_metadata(stream),
+                metadata={  # type: ignore
+                    **dagster_sling_translator.get_metadata(stream),
+                    METADATA_KEY_TRANSLATOR: dagster_sling_translator,
+                    METADATA_KEY_REPLICATION_CONFIG: replication_config,
```

Comment:
Just to get ahead of any memory issues here:

Right now, this will actually attach the serialized config in the asset. From experience with the dbt integration, we don't want that -- we just need this reference in user code, but not in the Dagster UI. We hide it anyways.

So we want something like the following PRs:
- https://github.com/dagster-io/dagster/pull/15447
- https://github.com/dagster-io/dagster/pull/17996

See more context here: https://github.com/dagster-io/dagster/pull/15447#discussion_r1478381482

Feel free to do this in this PR, or in a downstream one. But it should be done.

Reply:
Will create a Linear task to follow-up. 👍